react-app/src/account/Account.test.js

```diff
@@ -0,0 +1,52 @@
+import Account, { ErrorAuthenticationFail } from './Account.js';
+import naclUtils from 'tweetnacl-util';
+
+let account;
+
+const mockUserData = {
+  username:           '$/()RUFJ)wuwe84349',
+  password:           'dnefu384(/·$873hf7g',
+  expectedPublicKey:  '6FUygceUHlKzGu/5ir4FNkmAtUF07uNeAAdp9+5hcy4=',
+  expectedSecretKey:  'm4jBZuDiCgcCz94VIWHmeVC9IsXOIavniA2pqeq5Gg0='
+};
+
+test('match between generated key pairs with expected ones', () => {
+  expect.assertions(5);
+  account = new Account(mockUserData.username, mockUserData.password);
+
+  expect(account._username).toBe(mockUserData.username);
+
+  const systemTzOffsetInMinutes = -(new Date()).getTimezoneOffset();
+  const systemTzOffsetInHours = Math.floor(systemTzOffsetInMinutes / 60);
+  const systemTz = (systemTzOffsetInHours >= 0) ?
+    `GMT+${systemTzOffsetInHours}` :
+    `GMT${systemTzOffsetInHours}`;
+  expect(account._offset).toBe(systemTzOffsetInHours);
+  expect(account._tz).toBe(systemTz);
```

Comment:
Not sure there's a lot of value in checking the same code here as in the constructor. But I guess it doesn't hurt.

Reply:
Neither do I. It's the same situation with bytes() method, where test implies code-rewriting. I don't know which alternative is better, to have the guard and rewrite code or not to.

Reply:
If it's OK for all, I think that it's better to remove this guard (it contributes nothing) before merge. If the guard is OK, I have to add one for the bytes() method in order to be consistent.